botocore/handlers.py

```diff
@@ -596,6 +596,38 @@ def parse_get_bucket_location(parsed, http_response, **kwargs):
     parsed['LocationConstraint'] = region
 
 
+def document_s3_expires_shape(section, event_name, **kwargs):
+    # Updates the documentation for S3 operations that include the 'Expires' member
+    # in their response structure. Documents a synthetic member 'ExpiresString' and
+    # includes a deprecation notice for 'Expires'.
+    if 'response-example' in event_name:
+        if not section.has_section('structure-value'):
+            return
+        parent = section.get_section('structure-value')
+        if not parent.has_section('Expires'):
+            return
+        param_line = parent.get_section('Expires')
+        param_line.add_new_section('ExpiresString')
+        new_param_line = param_line.get_section('ExpiresString')
+        new_param_line.write("'ExpiresString': 'string',")
+        new_param_line.style.new_line()
+    elif 'response-params' in event_name:
+        if not section.has_section('Expires'):
+            return
+        param_section = section.get_section('Expires')
+        doc_section = param_section.get_section('param-documentation')
+        doc_section.write(
+            '*This member has been deprecated*. Please use ``ExpiresString`` instead.'
+        )
+        param_section.add_new_section('ExpiresString')
+        new_param = param_section.get_section('ExpiresString')
+        new_param.style.start_li()
+        new_param.write('**ExpiresString** (*string*) --')
+        new_param.style.end_li()
+        new_param.style.new_line()
+        new_param.write('\tThe raw, unparsed value of the ``Expires`` field.')
```

Comment:
Did the `\t` here come from somewhere else? I don't think that's actually rendered in the browser, it's going to have the same indentation as the above list won't it?

Actually, I'm also a little confused about the `end_li` before writing the last line. Looking at the live HTML, it looks like both values are within paragraph tags inside a single list element.

```html
<li>
<p><strong>Expires</strong> <em>(datetime) –</em></p>
<p>The date and time at which the object is no longer cacheable.</p>
</li>
```

Am I incorrectly conflating concepts? Do we have a sample of the actual HTML being generated?

Reply:
https://github.com/boto/botocore/pull/3232#discussion_r1725190777

Reply:
Sure, I've read the comment thread previously. What I'm asking about is the specific mechanics of the code. What's written doesn't make sense conceptually. Leading whitespace is not actually used in HTML, so there must be some other mechanism that is enforcing that tab before we render the page.

Reply:
Sure, I pulled this down and generated it real quick.  
With the `\t`: 
```
<ul>
<li><p><strong>ExpiresString</strong> (<em>string</em>) –</p>
<p>The raw, unparsed value of the <code class="docutils literal notranslate"><span class="pre">Expires</span></code> field.</p>
</li>
</ul>
```
without:
```
<ul class="simple">
<li><p><strong>ExpiresString</strong> (<em>string</em>) –</p></li>
</ul>
<p>The raw, unparsed value of the <code class="docutils literal notranslate"><span class="pre">Expires</span></code> field.</p>
```

That winds up being rendered out of alignment like this: 
![image](https://github.com/user-attachments/assets/a0332d50-ff4a-43c8-bda3-6e8306387cc0)

Looking at the full rendered HTML of the version with the `\t`, the list of parameters is kind of weirdly split into three unordered lists that don't have a separator between them (everything before `Expires`, `Expires`/`ExpiresString`, and everything after).  Not sure if that may render strangely in some contexts or be confusing for screen readers.

Reply:
The docs handler was fixed in a new commit. The HTML and RST now render how a modeled parameter would.